conans/client/conanfile/configure.py

```diff
@@ -32,6 +32,7 @@ def run_configure_method(conanfile, down_options, down_ref, ref):
         conanfile.settings.validate()  # All has to be ok!
         conanfile.options.validate()
         # Recipe provides its own name if nothing else is defined
+        # FIXME: This shouldn't be here
```

Comment:
Where instead? `init()` function closer to `set_name`?

Reply:
Yes, I think that can be defined at ``Conanfile`` class ``initialize()`` time, together with the ``requires``, ``settings`` and other stuff that is initialized very early in the recipe. At that point in time, the ``name`` attribute should be already defined.

Reply:
IIRC this is written here because the `provides` can be conditioned to options, e.g.: some recipe might provide `ssl` if an option like `vendorized_ssl`  is activated.

Reply:
This is the test: https://github.com/conan-io/conan/blob/aeb969369707005488df6d40785acecbffb5887c/conans/test/functional/provides/test_conditional_provides.py#L18

```suggestion
```